File: spark/spark1/src/main/java/com/examples/WordCount.scala
package com.examples

import org.apache.spark.sql.SparkSession
object WordCount {
  def main(args: Array[String]): Unit = {
    val spark = SparkSession
      .builder()
      .master("local[*]")//几个local代表几个分区，实际分区数由指定分区数量，默认值，核心数共同决定
      .appName("word count")
      .getOrCreate()
    val sc = spark.sparkContext
//    IDEA相对路径以项目根路径为准
    val rdd = sc.textFile("./data/word.txt")
    val counts = rdd.flatMap(_.split(" ")).map((_,1)).reduceByKey(_+_)
    counts.collect().foreach(println)
    println("全部的单词数："+counts.count())
    counts.saveAsTextFile("data/output/word-count")
  }
}

File: spark/spark1/src/main/java/com/operate/transformer/wordCount.java
package com.operate.transformer;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

import java.util.Arrays;

public class wordCount {
    public static void main(String[] args) {
        SparkConf sc = new SparkConf();
        sc.setMaster("local").setAppName("APP");
        JavaSparkContext jsc = new JavaSparkContext(sc);
        JavaRDD<String> lineRDD = jsc.textFile("./data/word.txt");
        JavaRDD<String> wordRDD = lineRDD.flatMap(line -> (Arrays.asList(line.split(" ")).iterator()));
        JavaPairRDD<String, Iterable<String>> groupRDD = wordRDD.groupBy(word -> word);
        JavaPairRDD<String, Integer> resRDD = groupRDD.mapValues(iter -> {
            int sum = 0;
            for (String s : iter) {
                sum++;
            }
            return sum;
        });
        resRDD.collect().forEach(System.out::println);
    }
}
